dc-commons-xml/src/main/java/de/digitalcollections/commons/xml/xpath/DocumentReader.java

```diff
@@ -244,6 +249,25 @@ private String resolveVariableContext(Locale language, String variableContext, M
     return result;
   }
 
+  private List<Element> resolveVariableAsElements(String[] paths) throws XPathMappingException {
+    paths = prependWithRootPaths(paths);
+    List<Element> result = new ArrayList<>();
+    for (String path : paths) {
+      List<Node> nodes;
+      try {
+        nodes = xpw.asListOfNodes(path);
+      } catch (IllegalArgumentException e) {
+        throw new XPathMappingException("Failed to resolve XPath: " + path, e);
+      }
+      for (Node node : nodes) {
+        if (node.getNodeType() == Node.ELEMENT_NODE) {
+          result.add((Element) node);
+        }
+      }
+    }
+    return result;
```

Comment:
Das kann man mit Streams ein bisschen eleganter machen IMO:

```suggestion
    try {
      return Arrays.stream(paths)
        .flatMap(path -> xpw.asListOfNodes(path).stream())
        .filter(node -> node.getNodeType() == Node.ELEMENT_NODE)
        .collect(Collectors.toList());
    } catch (IllegalArgumentException e) {
      throw new XPathMappingException("Failed to resolve XPath", e);
    } 
```

Reply:
done